### PR TITLE
feat(observability): add AppSurface OpenTelemetry package

### PR DESCRIPTION
**Pull request by akrock**
## Summary

- Added `ForgeTrust.AppSurface.Observability`, a public AppSurface package for app-side OpenTelemetry logging, tracing, and metrics registration.
- Wired the web example through `AppSurfaceObservabilityModule` so Aspire-hosted example runs can emit app telemetry to an OTLP collector.
- Updated the package chooser, readiness matrix, root/Aspire docs, release note, changelog, solution file, and lock files for the new package.

## Test Coverage

Coverage estimate from the ship audit: `91.09%` line, `80.76%` branch, `92.3%` method for `ForgeTrust.AppSurface.Observability`.

```mermaid
flowchart LR
  App["AppSurface app"] --> Module["AppSurfaceObservabilityModule"]
  Module --> Options["AppSurfaceObservabilityOptions"]
  Options --> Plan["AppSurfaceObservabilityPlan"]
  Plan --> Logs["OpenTelemetry logging"]
  Plan --> Traces["OpenTelemetry tracing"]
  Plan --> Metrics["OpenTelemetry metrics"]
  Plan --> Exporter{"OTLP endpoint or Always?"}
  Exporter -->|yes| OTLP["OTLP exporter"]
  Exporter -->|no| Diagnostic["no-endpoint startup diagnostic"]
  Aspire["Aspire / collector config"] --> Options
```

Covered paths:
- Config-source alignment for direct service and logging registration.
- Exporter mode/default resolution, endpoint precedence, environment fallback, and service identity override.
- First-registration-wins behavior for repeated registration.
- Module and example wiring compile paths.

Non-blocking gaps:
- OTLP exporter callback branches are indirectly covered by plan behavior, not direct callback assertions.
- Invalid configured `ExporterMode` and configured `OtlpEndpoint` branches are not direct tests; invalid environment endpoint is covered.
- Empty no-op methods remain uncovered and are intentionally low value.

## Pre-Landing Review

No blocking issues found. The earlier config-source drift concern is fixed and covered by regression tests. Repo-local `.agents/skills/gstack/review/checklist.md` was missing, so the ship review used the central gstack checklist at `/Users/andrew/gstack/.agents/skills/gstack/review/checklist.md`.

## Design Review

No frontend UI files changed. Design review skipped.

## Eval Results

No prompt-related files changed. Evals skipped.

## Scope Drift

Scope Check: CLEAN. The diff delivers the requested Aspire/OpenTelemetry observability hookup package, example wiring, and adoption docs.

## Plan Completion

No separate plan file detected for this ship run. Implementation, review, addressing, and release metadata were completed in-branch.

## Verification Results

- `dotnet test Observability/ForgeTrust.AppSurface.Observability.Tests/ForgeTrust.AppSurface.Observability.Tests.csproj` passed: 19 passed, 0 failed, 0 skipped.
- `dotnet test Aspire/ForgeTrust.AppSurface.Aspire.Tests/ForgeTrust.AppSurface.Aspire.Tests.csproj` passed: 13 passed, 0 failed, 0 skipped.
- `dotnet build examples/aspire-apphost/AspireAppHostExample.csproj` passed with 0 warnings and 0 errors.
- `dotnet run --project tools/ForgeTrust.AppSurface.PackageIndex/ForgeTrust.AppSurface.PackageIndex.csproj -- verify` passed.
- `dotnet format --verify-no-changes` passed.
- `git diff --check` passed.

Note: one initial parallel Aspire test attempt hit a transient compiler file lock while another build wrote `ForgeTrust.AppSurface.Core.dll`; the sequential rerun passed.

## TODOS

No `TODOS.md` exists in this repo, so no TODO items were marked complete.

## Documentation

- Added package README content for configuration shape, defaults, usage, pitfalls, and adoption boundaries.
- Updated root package discovery, Aspire docs, package chooser/readiness metadata, `releases/unreleased.md`, and `CHANGELOG.md`.

## Test plan

- [x] Observability package tests
- [x] Aspire package tests
- [x] Aspire AppHost example build
- [x] Package index verification
- [x] Formatting check
- [x] Whitespace check

### CODE REVIEW

File: Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityPlanTests.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    {
+        var exception = Assert.Throws<InvalidOperationException>(() =>
+            AppSurfaceObservabilityPlan.Resolve(
+                CreateContext("Catalog API"),
+                CreateConfiguration(),
+                environment: new TestEnvironmentReader(("OTEL_EXPORTER_OTLP_ENDPOINT", "not a uri"))));
+
+        Assert.Contains("OTEL_EXPORTER_OTLP_ENDPOINT must be an absolute URI", exception.Message);
+    }
+
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityPlanTests.cs:190]: Resolved on the current head: AppSurfaceOtlpExporterMode now pins explicit numeric values, and AppSurfaceOtlpExporterMode_NumericValuesAreStable covers the public enum contract.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityPlan.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        if (options.OtlpEndpoint is not null && !options.OtlpEndpoint.IsAbsoluteUri)
+        {
+            throw new InvalidOperationException(
+                $"{AppSurfaceObservabilityOptions.SectionName}:OtlpEndpoint must be an absolute URI.");
+        }
+
+        environment ??= AppSurfaceEnvironmentReader.Instance;
+        var endpoint = options.OtlpEndpoint
+            ?? ResolveUri(configuration[AppSurfaceObservabilityOptions.OtlpEndpointEnvironmentVariable])
+            ?? ResolveUri(environment.GetEnvironmentVariable(AppSurfaceObservabilityOptions.OtlpEndpointEnvironmentVariable));
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityPlan.cs:81]: Resolved on the current head: plan resolution now checks AppSurfaceObservability__OtlpEndpoint before the standard OTEL fallback, and the precedence tests cover that path.

File: Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityRegistrationTests.cs [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+        var plan = AppSurfaceObservabilityPlan.Resolve(
+            CreateContext("Catalog API"),
+            CreateConfiguration(),
+            environment: new TestEnvironmentReader());
+        var diagnostic = new AppSurfaceObservabilityStartupDiagnostic(plan, logger);
+
+        await diagnostic.StartAsync(CancellationToken.None);
+        await diagnostic.StartAsync(CancellationToken.None);
+
+        Assert.Equal(2, logger.Messages.Count);
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityRegistrationTests.cs]: Resolved on the current head: the startup diagnostic test now calls StartAsync once and asserts a single emitted message.

File: Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityPlanTests.cs [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        return new StartupContext([], new TestHostModule(), applicationName);
+    }
+
+    private static IConfiguration CreateConfiguration(params (string Key, string Value)[] values)
+    {
+        return new ConfigurationBuilder()
+            .AddInMemoryCollection(values.Select(value => new KeyValuePair<string, string?>(value.Key, value.Value)))
+            .Build();
+    }
+}
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityPlanTests.cs:260]: Resolved on the current head: plan tests now cover null context/configuration guards, default environment-reader handling, and blank/trimmed service metadata behavior.

File: Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityRegistrationTests.cs [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+            CreateContext("Catalog API"),
+            CreateConfiguration(("AppSurfaceObservability:ExporterMode", "Always")),
+            environment: new TestEnvironmentReader());
+
+        var builder = Sdk.CreateMeterProviderBuilder();
+        AppSurfaceObservabilityServiceCollectionExtensions.ConfigureMetrics(builder, plan);
+        using var provider = builder.Build();
+
+        Assert.NotNull(provider);
+    }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityRegistrationTests.cs:261]: Resolved on the current head: the tracing and metrics tests use the internal exporter callback seam and assert exporter registration is requested when the plan requires export.

File: Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityRegistrationTests.cs [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+            .ConfigureServices((_, services) =>
+            {
+                module.ConfigureHostBeforeServices(context, new DelegatingHostBuilder(services));
+                module.ConfigureHostAfterServices(context, new DelegatingHostBuilder(services));
+            })
+            .Build();
+
+        var options = host.Services.GetRequiredService<IOptions<AppSurfaceObservabilityOptions>>().Value;
+        Assert.Equal(AppSurfaceOtlpExporterMode.Never, options.ExporterMode);
+    }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityRegistrationTests.cs:368]: Resolved on the current head: the module hook test now verifies logging and service registration markers plus the hosted diagnostic registration, not just options binding.

File: Observability/ForgeTrust.AppSurface.Observability.Tests/DelegatingHostBuilder.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        var configuration = new ConfigurationBuilder()
+            .AddInMemoryCollection(new Dictionary<string, string?>
+            {
+                ["AppSurfaceObservability:ExporterMode"] = "Never"
+            })
+            .Build();
+        configureDelegate(new HostBuilderContext(Properties) { Configuration = configuration }, services);
+        return this;
+    }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability.Tests/DelegatingHostBuilder.cs:46]: Resolved on the current head: DelegatingHostBuilder now accepts optional configurationValues and only defaults ExporterMode to Never when no custom values are supplied.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceEnvironmentReader.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+{
+    string? GetEnvironmentVariable(string variable);
+}
+
+internal sealed class AppSurfaceEnvironmentReader : IAppSurfaceEnvironmentReader
+{
+    internal static readonly AppSurfaceEnvironmentReader Instance = new();
+
+    public string? GetEnvironmentVariable(string variable) => Environment.GetEnvironmentVariable(variable);
+}
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceEnvironmentReader.cs:37]: Resolved on the current head: the environment-reader interface, singleton implementation, and GetEnvironmentVariable method now include XML docs covering the test seam and null/blank handling contract.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityLoggingBuilderExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        options.IncludeScopes = true;
+        options.ParseStateValues = true;
+        options.SetResourceBuilder(ResourceBuilderFactory.Create(plan));
+
+        if (plan.ShouldRegisterExporter)
+        {
+            options.AddOtlpExporter(exporterOptions =>
+                AppSurfaceObservabilityServiceCollectionExtensions.ConfigureExporter(exporterOptions, plan));
+        }
+    }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityLoggingBuilderExtensions.cs:107]: Resolved on the current head: the internal logging overload and ConfigureLogging helper now document registerOptions, captured plan behavior, logging defaults, and exporter registration constraints.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityPlan.cs [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        if (string.IsNullOrWhiteSpace(value))
+        {
+            return null;
+        }
+
+        return Uri.TryCreate(value.Trim(), UriKind.Absolute, out var uri)
+            ? uri
+            : throw new InvalidOperationException(
+                $"{AppSurfaceObservabilityOptions.OtlpEndpointEnvironmentVariable} must be an absolute URI.");
+    }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityPlan.cs:129]: Resolved on the current head: AppSurfaceObservabilityPlan, Resolve, and ResolveUri now document parameters, exceptions, endpoint precedence, validation, and sharp edges.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityRegistrationMarkers.cs [hunk 11]
```diff
@@ -0,0 +1,5 @@
+namespace ForgeTrust.AppSurface.Observability;
+
+internal sealed class AppSurfaceObservabilityServicesRegistrationMarker;
+
+internal sealed class AppSurfaceObservabilityLoggingRegistrationMarker;
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityRegistrationMarkers.cs:21]: Resolved on the current head: both registration marker types now document first-registration-wins semantics, idempotency, and why logging and tracing/metrics markers stay separate.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityServiceCollectionExtensions.cs [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+    }
+
+    internal static void ConfigureResource(ResourceBuilder resource, AppSurfaceObservabilityPlan plan)
+    {
+        resource.AddService(
+            serviceName: plan.ServiceName,
+            serviceVersion: plan.ServiceVersion,
+            serviceInstanceId: null);
+    }
+}
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityServiceCollectionExtensions.cs:303]: Resolved on the current head: the service-collection helper APIs now include XML docs for idempotency, option binding, exporter/resource configuration, and the internal test seams.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityStartupDiagnostic.cs [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+internal sealed class AppSurfaceObservabilityStartupDiagnostic(
+    AppSurfaceObservabilityPlan plan,
+    ILogger<AppSurfaceObservabilityStartupDiagnostic> logger) : IHostedService
+{
+    public Task StartAsync(CancellationToken cancellationToken)
+    {
+        if (plan.ShouldLogSkippedExporterDiagnostic)
+        {
+            logger.LogInformation(
+                "AppSurface observability registered logging, tracing, and metrics without OTLP export because no endpoint was configured. Set {Section}:OtlpEndpoint, {Section}__OtlpEndpoint, or {OtlpEndpointEnvironmentVariable}, or set ExporterMode to Always or Never.",
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityStartupDiagnostic.cs]: Resolved on the current head: the diagnostic text now says tracing and metrics, and this older thread is outdated against the current diff.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceTelemetrySources.cs [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    /// Gets common .NET meter names that AppSurface opts into when they are emitted by the host runtime.
+    /// </summary>
+    public static readonly string[] StandardMeterNames =
+    [
+        MeterName,
+        "Microsoft.AspNetCore.Hosting",
+        "Microsoft.AspNetCore.Server.Kestrel",
+        "System.Net.Http",
+        "System.Runtime"
+    ];
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceTelemetrySources.cs]: Resolved on the current head: telemetry source lists are exposed as read-only collections and tests assert callers cannot mutate them.

File: Observability/ForgeTrust.AppSurface.Observability/ForgeTrust.AppSurface.Observability.csproj [hunk 15]
```diff
@@ -0,0 +1,21 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net10.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/ForgeTrust.AppSurface.Observability.csproj:8]: Resolved on the current head: the Observability project now sets GenerateDocumentationFile to true.

File: Observability/ForgeTrust.AppSurface.Observability/ResourceBuilderFactory.cs [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+internal static class ResourceBuilderFactory
+{
+    internal static ResourceBuilder Create(AppSurfaceObservabilityPlan plan)
+    {
+        return ResourceBuilder.CreateDefault()
+            .AddService(
+                serviceName: plan.ServiceName,
+                serviceVersion: plan.ServiceVersion,
+                serviceInstanceId: null);
+    }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/ResourceBuilderFactory.cs:25]: Resolved on the current head: ResourceBuilderFactory.Create now documents the OpenTelemetry defaults, shared resource identity behavior, plan parameter, and host extension point.

File: Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityStartupDiagnostic.cs [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+internal sealed class AppSurfaceObservabilityStartupDiagnostic(
+    AppSurfaceObservabilityPlan plan,
+    ILogger<AppSurfaceObservabilityStartupDiagnostic> logger) : IHostedService
+{
+    public Task StartAsync(CancellationToken cancellationToken)
+    {
+        if (plan.ShouldLogSkippedExporterDiagnostic)
+        {
+            logger.LogInformation(
+                "AppSurface observability registered tracing and metrics without OTLP export because no endpoint was configured. Set {Section}:OtlpEndpoint, {Section}__OtlpEndpoint, or {OtlpEndpointEnvironmentVariable}, or set ExporterMode to Always or Never.",
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability/AppSurfaceObservabilityStartupDiagnostic.cs]: Resolved in aa16acca: the no-endpoint startup diagnostic now says logging, tracing, and metrics, matching the package behavior and README scope.

File: Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityRegistrationTests.cs [hunk 18]
```diff
@@ -0,0 +1,411 @@
...
+        using var provider = services.BuildServiceProvider();
+        var options = provider.GetRequiredService<IOptions<AppSurfaceObservabilityOptions>>().Value;
+
+        Assert.Equal(AppSurfaceOtlpExporterMode.Never, options.ExporterMode);
+        Assert.Equal("orders-api", options.ServiceName);
+    }
+
+    [Fact]
+    public void ConfigureLogging_AddsExporterWhenPlanRequiresExport()
+    {
```
[the remark this answers is not in the document]
Comment (akrock) [reply Observability/ForgeTrust.AppSurface.Observability.Tests/AppSurfaceObservabilityRegistrationTests.cs:174]: Resolved in aa16acca: renamed the test to ConfigureLogging_SetsStructuredLoggingDefaults so it matches the assertions it actually makes.

File: packages/README.md [hunk 19]
```diff
@@ -59,6 +60,7 @@ Swipe to compare package details on narrow screens.
...
 | `ForgeTrust.AppSurface.Auth.AspNetCore.Oidc` | Add this in an ASP.NET Core web host that already chose an OpenID Connect provider and wants named AppSurface cookie/OIDC schemes with safe defaults and diagnostics. | `dotnet package add ForgeTrust.AppSurface.Auth.AspNetCore.Oidc` | AddAppSurfaceOidcAuth registration, AppSurfaceOidcAuthOptions, named cookie and OpenID Connect schemes, authorization-code defaults, SaveTokens false by default, local-only passive login/logout prompt helpers, OIDC subject mapping handoff, event chaining, and stable safe diagnostic codes. | User stores, ASP.NET Identity replacement, identity-provider hosting, OAuth/OIDC server behavior, provider SDKs, EF Core, persistence, middleware insertion, active redirects, challenge execution, sign-in or sign-out execution, silent default-scheme takeover, token storage by default, Aspire Keycloak, Dev Auth, or app-user provisioning. | [Package README](../Auth/ForgeTrust.AppSurface.Auth.AspNetCore.Oidc/README.md) | public preview<br />commercial ready<br />[notes](../releases/unreleased.md) |
```
[the remark this answers is not in the document]
Comment (akrock) [reply packages/README.md]: Resolved in aa16acca: changed the OIDC package chooser wording to AppSurface-named in package-index.yml and the generated packages/README.md row.
